hack/build/ci/generate-new-helm-index-yaml.sh

```diff
@@ -17,6 +17,8 @@ trap 'cleanup' ERR
 
 cp "config/helm/repos/stable/index.yaml" "config/helm/repos/stable/index.yaml.previous"
 
+mkdir -p "${output_dir}"
```

Comment:
why do we need this step?

Reply:
but if the folder doesn't exist it means the previous step failed and the helm index won't work
am I missing something?

Reply:
The foldes is passed as parameter and so far in master branch is empty = always failing ?
https://github.com/Dynatrace/dynatrace-operator/blob/main/.github/workflows/release.yaml#L327

Reply:
I tried to fix it with the changes of this PR

Reply:
I think it's another mess up with merge conflicts:
my version was https://github.com/andriisoldatenko/dynatrace-operator/blob/main/.github/workflows/release.yaml#L103

Reply:
Thanks! I will then use that as a reference :+1: